### PR TITLE
Save/restore blockface reservations

### PR DESCRIPTION
**Pull request by kdeloach**
This creates a reusable module for saving/loading stuff to localStorage and implements saving/loading state on the reserve blockfaces page.

### CODE REVIEW

File: src/nyc_trees/js/src/lib/storage.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+"use strict";
+
+var $ = require('jquery'),
+    storage = window.localStorage || {};
+
+
+function Storage(options) {
```
Comment (kdeloach) [inline src/nyc_trees/js/src/lib/storage.js]: I will probably rename this.

Comment (kdeloach) [inline src/nyc_trees/js/src/lib/storage.js]: TODO: Documentation

File: src/nyc_trees/js/src/lib/storage.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    }, options);
+
+    function save() {
+        storage[opts.key] = serialize();
+    }
+
+    function load() {
+        var state = deserialize(storage[opts.key]);
+        if (state) {
+            opts.validate(state);
```
Comment (maurizi) [inline src/nyc_trees/js/src/lib/storage.js]: Letting the exceptions from `validate` bubble up isn't the best thing...

Technically the `LocalStorage` could be edited by a user (though it seems like a low probability), or a bug in our code could make saved state no longer validate.

It's probably more robust to wrap this in a try/catch, and if there are any exceptions, log it and delete whatever was saved.

Comment (kdeloach) [inline src/nyc_trees/js/src/lib/storage.js]: I'm in favor of throwing here and letting the consumer decide how to cleanup the mess. I'll have to think about it.

**Comment by kdeloach** [comment]
@azavea-bot rebuild

**Comment by kdeloach** [comment]
I refactored the selectable blockface module to be its own Leaflet layer. Take another look when you get the chance.

**Comment by maurizi** [comment]
Since `selectableBlockfaceLayer.js` is now a class that gets instantiated with `new`, it should be renamed to start with a capital letter (`SelectableBlockfaceLayer.js`)

**Comment by maurizi** [comment]
We need to update [`reservationPage.js`](https://github.com/azavea/nyc-trees/blob/develop/src/nyc_trees/js/src/reservationPage.js#L33-L45) to accomodate the changes in `selectableBlockfaceLayer.js`

**Comment by maurizi** [comment]
Overall this looks really nice, modulo the 2 comments above.

+1 once those are addressed.

**Comment by kdeloach** [comment]
Done and done. Merging now.

**Pull request merged**